tests/test_storage.py

```diff
@@ -267,6 +268,140 @@ def test_get_orders(storage_mock):
+                    "createdBy": {
+                        "id": "1094497b-11d8-4fb8-9d6a-5e24a88aa825",
+                        "type": "USER",
+                    },
+                    "updatedBy": {"id": "system", "type": "INTERNAL"},
+                },
+            ],
+        ),
+        (
+            {
+                "workspace_orders": True,
+                "return_json": True,
+                "order_type": "TASKING",
+                "statuses": None,
+                "name": "Testing Name",
+            },
+            JSON_ORDERS,
+        ),
+        (
+            {
+                "workspace_orders": False,
+                "return_json": True,
+                "order_type": "TASKING",
+                "statuses": None,
+                "name": None,
+            },
+            JSON_ORDERS,
+        ),
+        (
+            {
+                "workspace_orders": False,
+                "return_json": True,
+                "order_type": "TASKING",
+                "statuses": None,
+                "name": None,
+            },
+            JSON_ORDERS,
+        ),
+    ],
+    ids=[
+        "Sc 1: statuses None, workspace_orders and return_json True, name None",
+        "Sc 2: statuses allowed, workspace_orders and return_json True, name None",
+        "Sc 3: statuses non allowed, workspace_orders and return_json True, name None",
+        "Sc 4: statuses allowed, workspace_orders True, return_json False, name None -> orders output expected",
+        "Sc 5: statuses None, workspace_orders and return_json True, name Test",
+        "Sc 6: Workspace Orders, Return JSON True, Statuses Test, name None",
+        "Sc 7: statuses None, workspace_orders False, Return JSON True, name None",
+    ],
+)
+def test_get_orders_v2_endpoint_params(
+    auth_mock, requests_mock, params, expected_output
+):
+    workspace_url_param = (
+        "&workspaceId=workspace_id_123" if params["workspace_orders"] else ""
+    )
+    url_storage_assets_paginated = f"{auth_mock._endpoint()}/v2/orders?sort=createdAt%2Cdesc{workspace_url_param}&type={params['order_type']}&size=50"
+    requests_mock.get(url=url_storage_assets_paginated, json=JSON_ORDERS)
+    expected_results = {}
+    if not params["return_json"]:
+        expected_results["content"] = [
+            Order(
+                auth=auth_mock,
+                order_id=output["id"],
+                order_info=output,
+            )
+            for output in expected_output
+        ]
+    else:
+        expected_results = expected_output
+
+    storage = Storage(auth=auth_mock)
+    orders = storage.get_orders(
+        order_type=params["order_type"],
+        statuses=params["statuses"],
+        name=params["name"],
+        workspace_orders=params["workspace_orders"],
+        return_json=params["return_json"],
+    )
+
+    for order, expected_result in zip(orders, expected_results["content"]):
+        assert repr(order) == repr(expected_result)
```

Comment:
I assert repr, because I wanted to assert in one line the json response as well as the object response.